tests/unittest_inference.py

```diff
@@ -5316,6 +5316,11 @@ def test_slice_inference_in_for_loops_not_working() -> None:
         assert inferred == util.Uninferable
 
 
+def test_slice_zero_step_does_not_raise_ValueError() -> None:
+    node = extract_node("[][::0]")
+    assert helpers.safe_infer(node) is None
```

Comment:
Preference on this integration with `safe_infer` vs isolation (and thus coupling to `InferenceError` behaviour instead of `Uniniferable`?)

Reply:
I'd prefer testing that this returns `Uninferable`.

Reply:
It does not return `Uninferable`, it raises `InferenceError`

Reply:
Oh, yeah of course. Let's test assigning this to a var then and inferring the var

Reply:
Like this?

```diff
 def test_slice_zero_step_does_not_raise_ValueError() -> None:
-    node = extract_node("[][::0]")
-    with pytest.raises(InferenceError):
-        next(node.infer())
+    node = extract_node("x = [][::0]; x")
+    assert next(node.infer()) == util.Uninferable
```

Reply:
Yeah!